File: MichiMark/Projection/EventDetail/MarkLinkItemProjection.swift
import Foundation

public struct MarkLinkItemProjection: Identifiable, Equatable {
    public let id: MarkLinkID
    public let markLinkSeq: Int//表示順
    public let markLinkType: MarkOrLink//マークorリンク
    
    public let displayDate: String//日付
    public let markLinkName: String//マークリンク名
    
    
    public let members: [MemberItemProjection]//メンバー
    
    public let displayMeterValue: String?//メーター
    public let displayDistanceValue: String?//走行距離
    public let actions: [ActionItemProjection]//行動
    public let isFuel: Bool
    public let pricePerGas: Int?/// 単位: 1円/L 通過型に変換
    public let gasQuantity: Double?/// 単位: 0.1L（10倍値）
    public let gasPrice: Int?/// 単位: 1円 通過型に変換
    public let memo: String?
}

extension MarkLinkItemProjection {

    static func emptry(markLinkId: MarkLinkID) -> Self {
        .init(
            id: markLinkId,
            markLinkSeq: 0,
            markLinkType: .mark,
            displayDate: "",
            markLinkName: "",
            members: [],
            displayMeterValue: nil,
            displayDistanceValue: nil,
            actions: [],
            isFuel: false,
            pricePerGas: nil,
            gasQuantity: nil,
            gasPrice: nil,
            memo: nil
        )
    }
    
    static func dummy(
        title: String = "自宅",
        date: String = "2026/01/01",
        markLink: MarkOrLink = .mark,
        seq: Int = 0
    ) -> Self {
        .init(
            id: MarkLinkID(),
            markLinkSeq: seq,
            markLinkType: markLink,
            displayDate: date,
            markLinkName: title,
            members: [],
            displayMeterValue: "1,000km",
            displayDistanceValue: "10.6km/L",
            actions: [],
            isFuel: false,
            pricePerGas: nil,
            gasQuantity: nil,
            gasPrice: nil,
            memo: ""
        )
    }

}

File: flutter/lib/features/shared/projection/mark_link_item_projection.dart
import 'package:equatable/equatable.dart';
import '../../../domain/transaction/mark_link/mark_or_link.dart';
import 'member_item_projection.dart';
import 'action_item_projection.dart';

class MarkLinkItemProjection extends Equatable {
  final String id;

  /// 表示順
  final int markLinkSeq;

  /// マーク / リンク 区分
  final MarkOrLink markLinkType;

  /// 日付の表示文字列（例: "2026/03/26"）
  final String displayDate;

  /// 名称
  final String markLinkName;

  /// 参加メンバー
  final List<MemberItemProjection> members;

  /// 累積メーターの表示文字列（例: "1,234 km"）。Mark用
  final String? displayMeterValue;

  /// 区間距離の表示文字列（例: "10.6 km"）。Link用
  final String? displayDistanceValue;

  /// 行動リスト
  final List<ActionItemProjection> actions;

  final bool isFuel;

  /// ガソリン単価（単位: 1円/L）
  final int? pricePerGas;

  /// 給油量（単位: 0.1L の実数値。例: 305 → 30.5）
  final double? gasQuantity;

  /// 給油金額（単位: 1円）
  final int? gasPrice;

  final String? memo;

  const MarkLinkItemProjection({
    required this.id,
    required this.markLinkSeq,
    required this.markLinkType,
    required this.displayDate,
    required this.markLinkName,
    required this.members,
    this.displayMeterValue,
    this.displayDistanceValue,
    required this.actions,
    required this.isFuel,
    this.pricePerGas,
    this.gasQuantity,
    this.gasPrice,
    this.memo,
  });

  @override
  List<Object?> get props => [
        id,
        markLinkSeq,
        markLinkType,
        displayDate,
        markLinkName,
        members,
        displayMeterValue,
        displayDistanceValue,
        actions,
        isFuel,
        pricePerGas,
        gasQuantity,
        gasPrice,
        memo,
      ];
}
